internal/store/leaf/crypto.go

```diff
@@ -162,14 +175,26 @@ func (s *Store) importPublicKey(ctx context.Context, r string) error {
 			continue
 		}
 		pk, err := s.storage.Get(ctx, filename)
-		if err != nil {
-			return err
-		}
+		return pk, err
+	}
+	return nil, fmt.Errorf("public key not found in store")
+}
+	
+// import an public key into the default keyring.
+func (s *Store) importPublicKey(ctx context.Context, r string) error {
+	im, ok := s.crypto.(keyImporter)
+	if !ok {
+		debug.Log("importing public keys not supported by %T", s.crypto)
 
-		return im.ImportPublicKey(ctx, pk)
+		return nil
 	}
 
-	return fmt.Errorf("public key not found in store")
+	pk, err := s.getPublicKey(ctx, r)
+	if err != nil {
+		return err
+	} else {
```

Comment:
You should just drop the else clause. If the if clause already returned the error then that case won't get there.

Example:

```
if err != nil {
  return err
}
return im.ImportPublicKey(ctx,pk)
```